beancount/utils/file_utils.py

```diff
@@ -101,6 +103,7 @@ def touch_file(filename, *otherfiles):
             break
 
 
+@typing_extensions.deprecated("use contextlib.chdir or community-backport instead")
 @contextlib.contextmanager
 def chdir(directory):
```

Comment:
Why not just remove chdir? If we don't use it elsewhere in this repo, remove.

Reply:
I don't see it used anywhere else in the beancount repo. I think it's safe to remove.

Reply:
> If we don't use it elsewhere in this repo, remove.

Someone is using it.

https://github.com/alexjj/money-scripts/blob/ddaee26d61092bee1db84e9e9034a20a39309a1b/bean-bake.py#L216

I have a general questions about this situation. I find there are severial thing in thie repo we don't use. For example, `date_utils.intimezone`. Should we just remove them?

Reply:
If it's used in another repo like this + we're on the master branch, we can remove it here. The beancount repo should contain only the minimum  it needs. That other repo can switch to using contextlib.chdir.

To your second question, yes, we should probably remove old things that aren't public and aren't in use anymore, but it's not super important. Those exist because all the bean* repos used to be in here, and they were forked to other repositories under github.com/beancount. Now, it might be worthwhile to move that code to those repos if it's still used there, except maybe if they're used broadly in a bunch of repos and related to beancount functionality, in which case we'd consider them publicly provided by this repo.  

snoop: This is useful for regexps but don't use it anymore (it was used by bean-bake iirc). We can remove.

intimezone: Can also remove if it's not used. This was used for some tests (surprised we're not using it anymore).

Reply:
Also, it's best to do these removals in dedicated PRs each, so we have a history of them.

Reply:
ok, done

Reply:
> intimezone: Can also remove if it's not used. This was used for some tests (surprised we're not using it anymore).

I guess the reason is it can't work on windows